<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title> 2 TAKSONOMI FILOGENETIK | Modul Praktikum Sistematika Mikrobia</title>
  <meta name="description" content=" 2 TAKSONOMI FILOGENETIK | Modul Praktikum Sistematika Mikrobia" />
  <meta name="generator" content="bookdown 0.25 and GitBook 2.6.7" />

  <meta property="og:title" content=" 2 TAKSONOMI FILOGENETIK | Modul Praktikum Sistematika Mikrobia" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="https://danilyanedo7.github.io/praktikumsismikbits/images/cover.png" />
  
  <meta name="github-repo" content="danilyanedo7/praktikumsismikbits" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content=" 2 TAKSONOMI FILOGENETIK | Modul Praktikum Sistematika Mikrobia" />
  
  
  <meta name="twitter:image" content="https://danilyanedo7.github.io/praktikumsismikbits/images/cover.png" />



<meta name="date" content="2022-04-08" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="daftar-pustaka.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/tabwid-1.0.0/tabwid.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<script src="libs/plotly-binding-4.10.0/plotly.js"></script>
<script src="libs/typedarray-0.1/typedarray.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="libs/plotly-htmlwidgets-css-2.5.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="libs/plotly-main-2.5.1/plotly-latest.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="./">Modul Praktiikum Sismik</a></li><img src="https://www.its.ac.id/biologi/wp-content/uploads/sites/21/2018/08/LOGO-BIO-PUTIH.png"></a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> TAKSONOMI NUMERIK FENETIK<span></span></a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#klasifikasi-numerik-fenetik-berdasarkan-data-fenotipik"><i class="fa fa-check"></i><b>1.1</b> Klasifikasi Numerik-Fenetik Berdasarkan Data Fenotipik<span></span></a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="index.html"><a href="index.html#pengantar"><i class="fa fa-check"></i><b>1.1.1</b> Pengantar<span></span></a></li>
<li class="chapter" data-level="1.1.2" data-path="index.html"><a href="index.html#cara-kerja"><i class="fa fa-check"></i><b>1.1.2</b> Cara Kerja<span></span></a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#klasifikasi-numerik-fenetik-berdasarkan-profil-sidik-jari-molekular"><i class="fa fa-check"></i><b>1.2</b> Klasifikasi Numerik-Fenetik Berdasarkan Profil Sidik Jari Molekular<span></span></a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#pengantar-1"><i class="fa fa-check"></i><b>1.2.1</b> Pengantar<span></span></a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#cara-kerja-1"><i class="fa fa-check"></i><b>1.2.2</b> Cara Kerja<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="taksonomi-filogenetik.html"><a href="taksonomi-filogenetik.html"><i class="fa fa-check"></i><b>2</b> TAKSONOMI FILOGENETIK<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="taksonomi-filogenetik.html"><a href="taksonomi-filogenetik.html#klasifikasi-molekular-filogenetik-berdasarkan-gen-16s-rrna"><i class="fa fa-check"></i><b>2.1</b> Klasifikasi Molekular Filogenetik Berdasarkan Gen 16S rRNA<span></span></a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="taksonomi-filogenetik.html"><a href="taksonomi-filogenetik.html#pengantar-2"><i class="fa fa-check"></i><b>2.1.1</b> Pengantar<span></span></a></li>
<li class="chapter" data-level="2.1.2" data-path="taksonomi-filogenetik.html"><a href="taksonomi-filogenetik.html#cara-kerja-2"><i class="fa fa-check"></i><b>2.1.2</b> Cara Kerja<span></span></a></li>
<li class="chapter" data-level="2.1.3" data-path="taksonomi-filogenetik.html"><a href="taksonomi-filogenetik.html#konstruksi-pohon-filogeni-menggunakan-software-mega-4.0"><i class="fa fa-check"></i><b>2.1.3</b> Konstruksi Pohon Filogeni Menggunakan Software MEGA 4.0<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="daftar-pustaka.html"><a href="daftar-pustaka.html"><i class="fa fa-check"></i><b>3</b> DAFTAR PUSTAKA<span></span></a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Modul Praktikum Sistematika Mikrobia</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="taksonomi-filogenetik" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number"> 2</span> TAKSONOMI FILOGENETIK<a href="taksonomi-filogenetik.html#taksonomi-filogenetik" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="klasifikasi-molekular-filogenetik-berdasarkan-gen-16s-rrna" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Klasifikasi Molekular Filogenetik Berdasarkan Gen 16S rRNA<a href="taksonomi-filogenetik.html#klasifikasi-molekular-filogenetik-berdasarkan-gen-16s-rrna" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="pengantar-2" class="section level3 hasAnchor" number="2.1.1">
<h3><span class="header-section-number">2.1.1</span> Pengantar<a href="taksonomi-filogenetik.html#pengantar-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Klasifikasi molekular filogenetik merupakan klasifikasi yang disusun dengan mempertimbangkan jalur evolusi setiap organisme yang dikaji. Hal ini berbeda dengan klasifikasi fenetik yang hanya melihat hubungan antar organisme berdasarkan karakter yang ada pada saat ini (Priest &amp; Austin, 1993). Dalam prosesnya, klasifikasi molekular filogenetik menggunakan data berupa urutan (sequence) nukleotida pada DNA atau asam amino pada protein. Sequence nukleotida yang digunakan dalam klasifikasi molekular filogenetik harus merupakan sequence yang diwariskan langsung oleh nenek moyang (homolog) serta memiliki kesamaan sejarah evolusi. Sebuah sequence dapat disebut sebagai marker molekular apabila memenuhi persyaratan berupa: (1) terdistribusi pada seluruh organisme, (2) memiliki kesetaraan fungsi pada seluruh organisme, dan (3) memegang peranan vital bagi kehidupan organisme. Hal ini menjadikan marker molekular merupakan sequence yang tepat untuk digunakan dalam studi klasifikasi filogenetik.</p>
<p>Gen 16S rRNA merupakan salah satu contoh marker molekular karena terdapat pada organisme baik yang berada pada domain Bacteria, Archaea, serta Eukarya. Saat ini, informasi mengenai sequence gen 16S rRNA sudah sangat banyak tersedia pada database internasional dan juga sudah dijadikan standar penetapan suatu spesies baru dalam studi taksonomi. Pada praktikum ini kita akan mencoba mengklasifikasikan sejumlah bakteri dengan cara merekonstruksi pohon filogenetik berdasarkan sequence gen 16S RNA yang dimiliki.</p>
</div>
<div id="cara-kerja-2" class="section level3 hasAnchor" number="2.1.2">
<h3><span class="header-section-number">2.1.2</span> Cara Kerja<a href="taksonomi-filogenetik.html#cara-kerja-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="persiapan-data" class="section level4 hasAnchor" number="2.1.2.1">
<h4><span class="header-section-number">2.1.2.1</span> Persiapan Data<a href="taksonomi-filogenetik.html#persiapan-data" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Data yang digunakan dalam klasifikasi berbasis molekular filogenetik adalah berupa urutan (sequence) nukleotida atau asam amino. Kedua jenis sequence ini umumnya dapat diperoleh dari hasil sequencing DNA/protein maupun dari sequence database internasional yang tersedia di internet. Untuk kemudahan dalam praktikum ini, kita akan menggunakan cara kedua, yakni mengunduh sequence nukleotida/asam amino dari internet. Saat ini sudah banyak sekali situs yang memfasilitasi hal tersebut. Situs GenBank, DDBJ, serta EMBL merupakan situs dimana pencarian sequence nukleotida/asam amino umumnya dilakukan. Dalam acara praktikum ini, kita akan mencoba mengunduh data melalui GenBank (<a href="http://www.ncbi.nlm.nih.gov/Genbank/" class="uri">http://www.ncbi.nlm.nih.gov/Genbank/</a>) dan untuk selanjutnya sequence DNA akan digunakan sebagai contoh persiapan data.</p>
<p>Pada GenBank kita dapat mencari sequence DNA berdasarkan nama gen, spesies pemilik gen, atau accession number pada kolom search yang tersedia. Accession number merupakan penanda/identitas dari setiap sequence DNA yang telah disimpan pada situs tersebut. Pada umumnya accession number tertulis pada artikel publikasi ilmiah yang penelitinya telah menyumbangkan hasil sequencing-nya ke database yang ada. Pencarian pada menu search di GenBank terdiri dari banyak menu sub-pencarian untuk memudahkan penelusuran data, namun dalam praktikum ini kita akan lebih fokus pada pencarian dengan menu Nucleotide, Protein, dan Genome. Sebagai contoh, apabila kita ingin melakukan pencarian gen 16S rRNA bakteri Escherichia coli maka kita dapat melakukan salah satu dari hal berikut:
1. Melakukan penelusuran pustaka, misalnya pada artikel jurnal untuk mendapatkan accession number. Sebagai contoh, Escherichia coli ATCC 11775T pada Bergey’s Manual of Systematic Bacteriology (Brenner et al., 2005) memiliki accession number X80725. Accession Number ini dapat diketikan pada kolom search nucleotide. Cara ini merupakan pencarian yang cukup spesifik karena setiap sequence dari setiap jenis organisme memiliki accession number tersendiri.
2. Melakukan penelusuran pada situs List of Prokaryotic names with Standing in Nomenclature (LPSN, <a href="http://www.bacterio.cict.fr/" class="uri">http://www.bacterio.cict.fr/</a>). Penelusuran melalui situs ini khusus untuk pencarian sequence bakteri dan archaea saja. Informasi yang diberikan mengenai suatu genus/spesies meliputi type species, type strain, tahun ditemukan, dan publikasi (valid/efektif) terkait genus/spesies tersebut.
3. Melakukan penelusuran pada kolom search nucleotide di GenBank dengan mengetik kata kunci, seperti “Escherichia coli 16S rRNA”. Cara ini kurang spesifik karena akan menghasilkan sejumlah daftar yang berisikan berbagai macam spesies bakteri yang mengandung sequence gen 16S rRNA dan kita harus mencarinya satu per satu.
4. Melakukan penelusuran pada kolom search genome di GenBank dan kemudian mengetik nama spesies yang kita inginkan. Umumnya hasil penelusuran meliputi sejumlah daftar complete genome dari spesies yang dicari. Kita dapat mencari gen yang diinginkan dalam daftar complete genome tersebut. Cara ini cukup memakan waktu, namun kita dapat menelusuri berbagai jenis gen dari spesies tersebut.</p>
<p>Pada genbank, halaman tersebut akan mengandung informasi mengenai jenis sequence, asal sequence, produk yang dihasilkan, dan sequence itu sendiri. Dalam contoh ini jenis sequence adalah gen 16S rRNA yang berasal dari Escherichia coli ATCC 11775T. Gen ini akan menghasilkan produk berupa RNA ribosomal (rRNA). Informasi mengenai hasil pengkodean dari sequence dapat dilihat pada FEATURES. Sequence dari DNA itu sendiri yang dituliskan per 10 nukleotida pada kolom ORIGIN. Sequence ini selanjutnya dapat diunduh dalam bentuk FASTA dengan cara meng-klik link FASTA pada baris ketiga, dan akan muncul:</p>
<p>E.coli (ATCC 11775T) gene for 16S rRNA GenBank: X80725.1
GenBank Graphics
’&gt;gi|1240022|emb|X80725.1| E.coli (ATCC 11775T) gene for 16S rRNA AGTTTGATCATGGCTCAGATTGAACGCTGGCGGCAGGCCTAACACATGCAAGTCGAACGG TAACAGGAAGCAGCTTGCTGCTTTGCTGACGAGTGGCGGACGGGTGAGTAATGTCTGGGA AACTGCCTGATGGAGGGGGATAACTACTGGAAACGGTAGCTAATACCGCATAACGTCGCA AGCACAAAGAGGGGGACCTTAGGGCCTCTTGCCATCGGATGTGCCCAGATGGGATTA…</p>
<p>Kedua jenis data ini (informasi sequence dan sequence FASTA) dikompilasi ke masing-masing arsip untuk membuat database sendiri.</p>
<ol style="list-style-type: decimal">
<li>Sequence Alignment dengan Program ClustalX 2.1 (Larkin et al., 2001)
Alignment bertujuan untuk menata sequence agar satu sama lain diletakkan sesuai dengan posisi homologi antar sequence. Hanya berdasarkan alignment inilah kita dapat membandingkan antar sequence gen 16S rRNA dari masing-masing strain mikrobia yang akan diklasifikasikan. Alignment menggunakan program ClustalX dilakukan dengan mempersiapkan data sequence dalam format FASTA. Dataset sejumlah sequence yang telah didapatkan dikumpulkan terlebih dahulu ke dalam 1 file Notepad dengan awalan dari setiap sequence diberikan tanda “&gt;”. Contohnya dapat dilihat pada sequence berikut:</li>
</ol>
<p>’&gt;Allochromatium vinosum DSM 180 agagtttgatcctggctcagattgaacgctggcggcatgcctaacacatgcaa…</p>
<p>’&gt;Chlorobium luteolum DSM 273 aggaaagcggcttcggccgggagtacttggcgcaagggtgagtaaggcatagg…</p>
<p>’&gt;Chloroflexus aurantiacus J-10-fl aaaggaggtgatccagccgcaccttccggtacggctaccttgttacgacttcg…</p>
<p>Penamaan dalam pembuatan dataset sequence ini juga perlu diperhatikan. Disarankan untuk menyingkat nama dari setiap sequence yang ada karena program ClustalX akan secara otomatis memotong karakter nama apabila melebihi 30 karakter. Selain itu, hasil alignment ClustalX yang akan digunakan dalam program Phylip (.phy) juga akan secara otomatis memotong karakter nama apabila melebihi 10 karakter. Hal ini akan membingungkan apabila pembeda antar nama sequence satu dengan lainnya terletak pada posisi karakter &gt;10, karena ketika dipotong akan menghasilkan nama yang sama antar sequence satu dengan lainnya. Dengan demikian, sebaiknya kita membuat daftar baru (dengan MS Word atau Notepad) yang berisi rincian nama sequence beserta singkatannya. Singkatan nama ini yang akan kita gunakan dalam dataset yang akan diproses dalam ClustalX. Contoh dataset diatas setelah namanya disingkat akan menjadi:
’&gt;AvinDSM180
agagtttgatcctggctcagattgaacgctggcggcatgcctaacacatgcaa…</p>
<p>’&gt;ClutDSM273
aggaaagcggcttcggccgggagtacttggcgcaagggtgagtaaggcatagg…</p>
<p>’&gt;ChaurJ-10-fl aaaggaggtgatccagccgcaccttccggtacggctaccttgttacgacttcg…</p>
<p>Perlu diperhatikan bahwa pemberian nama tidak boleh mengandung spasi. Penyingkatan nama sequence hingga satu karakter (A-Z) sebaiknya tidak dilakukan karena akan membuat program ClustalX salah mengenali karakter nama menjadi salah satu komponen sequence nukleotida atau asam amino.</p>
<p>Jadi, contoh seperti:
’&gt;A
caaaatggagagtttgatcctggctcaggatgaacgctggcggcgtgcttaac…</p>
<p>tidak boleh dilakukan karena nama “A” akan disalahartikan sebagai nukleotida oleh ClustalX. Dataset yang telah siap selanjutnya disimpan dan selanjutnya dimasukan ke dalam ClustalX. Pertama buka program ClustalX, File → Load Sequences. Pilih file dataset yang telah disimpan dan program tersebut akan secara otomatis menampilkan nama sequence pada kolom sebelah kiri dan sequence-nya di kolom sebelah kanan.</p>
<p>Sebelum melakukan alignment, kita harus terlebih dahulu mengatur file output yang akan dihasilkan. Program ClustalX dapat menghasilkan beberapa macam format file dari satu jenis dataset, yakni: CLUSTAL, GCG/MSF, GDE, NBRF/PIR, NEXUS dan PHYLIP. Masing-masing file output ini mempunyai kegunaan masing-masing karena diperlukan oleh program-program analisis lainnya. Dalam praktikum ini kita hanya akan menggunakan format CLUSTAL (.aln), GDE (.gde), dan PHYLIP (.phy). Pada program ClustalX, pilih Alignment → Output Format Options → klik GDE FORMAT, dan PHYLIP FORMAT → CLOSE. Setelah itu pilih Alignment → Do Complete Alignment.</p>
</div>
<div id="rekonstruksi-pohon-filogenetik-dengan-phylip-v3.6.9-felsenstein-2005" class="section level4 hasAnchor" number="2.1.2.2">
<h4><span class="header-section-number">2.1.2.2</span> Rekonstruksi Pohon Filogenetik dengan PHYLIP v3.6.9 (Felsenstein, 2005)<a href="taksonomi-filogenetik.html#rekonstruksi-pohon-filogenetik-dengan-phylip-v3.6.9-felsenstein-2005" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Rekonstruksi dengan program ini membutuhkan file input dalam format .phy yang dapat dihasilkan oleh program ClustalX atau MEGA. Program Phylip memiliki serangkaian aplikasi executeable yang dapat menganalisis data sequence dengan berbagai algoritme. Aplikasi yang terdapat pada Phylip antara lain:</p>
<p>clique dnamlk drawgram neighbor restml
concense dnamove drawtree pars retree
contml dnapars factor penny seqboot
contrast dnapenny fitch proml treedist
dnacomp dollop gendist promlk<br />
dnadist dolmove main protdist<br />
dnainvar dolpenny mix protpars<br />
dnaml draw move restdist</p>
<p>Rekonstruksi pohon filogenetik pada Phylip dapat dilakukan dengan aplikasi neighbor, fitch, dnapars, dan dnaml untuk sequence nukleotida. Aplikasi neighbor juga dapat digunakan untuk merekonstruksi pohon dengan sequence asam amino, namun sebelumnya harus menggunakan input data yang dihasilkan dari aplikasi protdist. Aplikasi protml dan protpars memiliki prinsip analisis yang sama dengan dnaml dan dnapars, namun menggunakan sequence asam amino sebagai data inputnya.
Copy file format .phy ke dalam folder exe yang terdapat di dalam folder Phylip dan kemudian ganti nama file tersebut menjadi infile tanpa menggunakan extension/format apapun. Folder exe terkadang sudah mengandung file dengan nama infile yang disebabkan oleh analisis data yang pernah dilakukan sebelumnya. Apabila kita menjumpai file infile tersebut, hapus terlebih dahulu dan kemudian baru mengganti nama file .phy menjadi infile. Selanjutnya, buka salah satu dari aplikasi yang tersedia dalam folder exe untuk menganalisis infile, dalam hal ini kita akan menggunakan aplikasi neighbor, yakni rekonstruksi dengan algoritme Neighbor- Joining.
Perlu diketahui bahwa rekonstruksi pohon filogenetik yang menggunakan data distance matrix berbasis clustering seperti UPGMA, Neighbor-Joining, dan algoritme Fitch-Margoliash menggunakan data berupa matriks p-distance yang dihasilkan oleh aplikasi dnadist untuk sequence DNA atau protdist untuk sequence asam amino.</p>
<p>Dengan demikian, infile perlu dianalisis terlebih dahulu dengan dnadist, kemudian outfile hasil analisis dnadist diubah lagi (rename) menjadi infile untuk dapat dianalisis dengan neighbor. Setelah namanya diubah, buka aplikasi neighbor dan akan tampil menu berupa:
Neighbor-Joining/UPGMA method version 3.69 Settings for this run:
N Neighbor-joining or UPGMA tree? Neighbor-joining O Outgroup root? No, use as outgroup species 1
L Lower-triangular data matrix? No R Upper-triangular data matrix? No S Subreplicates? No
J Randomize input order of species? No. Use input order M Analyze multiple data sets? No
0 Terminal type (IBM PC, ANSI, none)? ANSI
1 Print out the data at start of run No
2 Print indications of progress of run Yes
3 Print out tree Yes
4 Write out trees onto tree file? Yes
Y to accept these or type the letter for one to change</p>
<p>Tekan huruf yang bersangkutan untuk mengubah pengaturan yang diinginkan. Setelah selesai, tekan Y dan Enter. File hasil analisis akan dimuat ke dalam outfile dan gambar pohon dimuat pada outtree. File outfile dapat dibuka dengan Notepad dan dapat di-copy ke dalam MS Excel untuk memudahkan pembacaan, sedangkan file outtree dapat dibuka dengan program Treeview (Page, 1996).
Rekonstruksi pohon filogenetik dengan algoritme berbasis character-based seperti maximum parsimony dan maximum likelihood menggunakan file .phy sebagai infile dan tidak perlu dibuat matriks p-distance. Dengan demikian kita dapat langsung menjalankan program dnapars untuk analisis parsimony atau dnaml untuk analisis likelihood langsung terhadap infile.</p>
<p>3.1.3 Pembuatan Matriks Similaritas DNA dengan Phydit (Chun, 1995)
Program Phydit menggunakan data input berupa hasil alignment dengan format .gde yang dapat dihasilkan dengan program ClustalX. Salah satu analisis yang dilakukan oleh Phydit adalah penghasilan matriks similaritas nukleotida yang berisi persentase similaritas nukleotida antar pasangan sequence yang dibandingkan tanpa menggunakan model evolusi apapun.</p>
<p>Analisis dengan Phydit dilakukan dengan membuat file baru dengan membuka menu File → New atau dengan menekan ikon New Phydit File yang terdapat pada panel atas bagian paling kiri. Sebuah menu akan muncul dimana kita dapat memasukan keterangan file. Isi keterangan apabila diperlukan dan apabila sudah, tekan OK. Phydit kemudian akan menampilkan layar baru bertuliskan No entry to tag. Pada tahap ini, masukan data melalui Data → Import → GDE (NT Replace) untuk sequence nukleotida. Pilih data dalam format .gde dan Phydit akan langsung memasukan sequence berdasarkan entry nama sequence. Penghasilan matriks similaritas nukleotida dapat dilakukan melalui Analysis → SimTable: Generating Similarity Table yang terdapat pada panel atas. Matriks similaritas nukleotida terdiri atas dua bagian, yakni bagian segitiga kanan atas (upper-right triangle) dan segitiga kiri bawah (lower-left triangle). Phydit akan menanyakan jenis data yang akan dimuat dalam masing-masing segitiga tersebut. Pada umumnya kita akan memasukan data similaritas nukleotida (NT Similarity) pada lower-left triangle dan jumlah nukleotida yang berbeda per total nukleotida yang dibandingkan (NT different/Total Nucleotides) pada upper-right triangle. Tekan OK dan akan muncul menu Options; tekan OK lagi untuk melanjutkan analisis. Hasil yang keluar berupa matriks yang dituliskan pada Notepad. Untuk memudahkan pembacaan, copy seluruh tulisan pada Notepad tersebut (ctrl+A kemudian ctrl+C) ke MS Excel dan kemudian simpan.
Sebagai tambahan, Phydit juga menyediakan pilihan analisis lainnya yang mencakup:
• Alignment Reports, menu ini berfungsi untuk menampilkan hasil alignment sequence
nukleotida/asam amino dalam bentuk text yang secara otomatis dibuka oleh Notepad.
• Sequence Statistics, menu ini berfungsi untuk menampilkan frekuensi nukleotida dan frekuensi asam amino dari sequence yang ada. Apabila kita menggunakan sequence nukleotida, adanya statistik frekuensi asam amino pada hasil diasumsikan bahwa sequnce yang dimiliki merupakan coding sequence.</p>
</div>
</div>
<div id="konstruksi-pohon-filogeni-menggunakan-software-mega-4.0" class="section level3 hasAnchor" number="2.1.3">
<h3><span class="header-section-number">2.1.3</span> Konstruksi Pohon Filogeni Menggunakan Software MEGA 4.0<a href="taksonomi-filogenetik.html#konstruksi-pohon-filogeni-menggunakan-software-mega-4.0" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Molecular Evolutionary Genetic Analysis 4.0 (MEGA 4.0) merupakan alat terintegerasi untuk mengolah pemerataan sekuen, menkonstruksi pohon filogenetik, memperkirakan perbedaan waktu, memperkirakan kecepatan evolusi molekuler, menkonstruksi sekuen nenek moyang, dan menguji hipotesis evolusioner. Aplikasi MEGA 4.0 digunakan oleh para ahli biologi di dalam laboratorium untuk merekonstruksi sejarah evolusioner dari suatu spesies dan sifat selektif yang dihasilkan dari kekuatan alam membentuk evolusi gen dan spesies.</p>
<p>Langkah-langkah merekonstruksi pohon filogenetik dengan aplikasi MEGA 4.0 lebih singkat daripada dengan cara diatas. Secara garis besar tahapan rekonstruksi pohon filogenetik dengan menggunakan aplikasi MEGA 4.0 terdiri atas:
1. Mengunduh sekuens nukleotida yang kita kehendaki dari GenBank
2. Sequence alignment dengan ClustalW
3. Rekonstruksi pohon filogenetik</p>
<div id="mengunduh-sequence-nukleotida-dari-genbank" class="section level4 hasAnchor" number="2.1.3.1">
<h4><span class="header-section-number">2.1.3.1</span> Mengunduh Sequence Nukleotida dari GenBank<a href="taksonomi-filogenetik.html#mengunduh-sequence-nukleotida-dari-genbank" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Pengunduhan dapat dilakukan dari NCBI (<a href="http://www.ncbi.nlm.nih.gov/Genbank" class="uri">http://www.ncbi.nlm.nih.gov/Genbank</a>) dengan format FASTA. Diupayakan sekuens nukelotida yang kita gunakan berasal dari jenis yang sama dan banyaknya sekuen nukleotida tidak jauh berbeda. Apabila kita memilih sekuen dengan panjang yang perbedaan terlalu jauh maka akan mengganggu proses alignment dengan ClustalW. Setelah mengunduh format FASTA dari GenBank kemudian disalin pada program Notepad.</p>
<ol style="list-style-type: decimal">
<li>Sequence alignment dengan ClustalW</li>
</ol>
<ul>
<li>Buka aplikasi MEGA 4.0, pilih menu Alignment kemudian beri tanda pada pilihan
Create new alignment dan klik OK.</li>
<li>Data sekuens dalam format FASTA dari Notepad kita salin pada blank sequence dengan cara Ctrl+N untuk menambak kotak blank sequence. Nama Sequence 1, Sequence 2, dst. dapat diganti dengan nama strain asli. Pada kotak sebelah nama strain ditempelkan data sekuen nukleotida. Basa T dengan kotak warna merah, basa G dengan kotak warna ungu, basa A dengan kotak warna hijau, dan basa C dengan kotak biru.</li>
<li>Setelah selesai melakukan penambahan sekuens kemudian klik icon W yang merupakan ClustalW. Hasil alignment dengan ClustalW disimpan dengan format file .mega.</li>
</ul>
</div>
<div id="rekonstruksi-pohon-filogenetik" class="section level4 hasAnchor" number="2.1.3.2">
<h4><span class="header-section-number">2.1.3.2</span> Rekonstruksi Pohon Filogenetik<a href="taksonomi-filogenetik.html#rekonstruksi-pohon-filogenetik" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>Buka kembali software MEGA 4, pilih File, kemudian pilih Reopen data, pilih data hasil alignment yang telah dibuat. Pilih menu Phylogeny, kemudian pilih Boostrap Test of Phylogeny, pilih Neighbor Joining (perlu diperhatikan jumlah boostrap sebaiknya 1000 replikasi), dan klik Compute.</li>
<li>Pohon filogenetik selesai dibuat.</li>
</ul>
</div>
<div id="rekonstruksi-pohon-filogenetik-dengan-r" class="section level4 hasAnchor" number="2.1.3.3">
<h4><span class="header-section-number">2.1.3.3</span> Rekonstruksi Pohon Filogenetik dengan R<a href="taksonomi-filogenetik.html#rekonstruksi-pohon-filogenetik-dengan-r" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Dalam proses rekonstruksi phohon filogenetik lebih lanjut menggunakan R akan menggunakan packages ggtree. Sumber utama dapat dilihat pada link berikut <a href="https://yulab-smu.top/treedata-book/" class="uri">https://yulab-smu.top/treedata-book/</a></p>
<p>Proses persiapan yaitu melakukan instalasi package. Proses install packages di bawah ini dengan menghapus tanda “#”, kemudian <em>run</em> code tersebut.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="taksonomi-filogenetik.html#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># install.packages(&quot;tidyverse&quot;)</span></span>
<span id="cb11-2"><a href="taksonomi-filogenetik.html#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="co"># install.packages(&quot;here&quot;)</span></span>
<span id="cb11-3"><a href="taksonomi-filogenetik.html#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="co"># install.packages(&quot;remotes&quot;)</span></span>
<span id="cb11-4"><a href="taksonomi-filogenetik.html#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="co"># remotes::install_github(&quot;GuangchuangYu/treeio&quot;)</span></span>
<span id="cb11-5"><a href="taksonomi-filogenetik.html#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="co"># install.packages(&quot;BiocManager&quot;)</span></span>
<span id="cb11-6"><a href="taksonomi-filogenetik.html#cb11-6" aria-hidden="true" tabindex="-1"></a><span class="co"># BiocManager::install(&quot;ggtree&quot;)</span></span>
<span id="cb11-7"><a href="taksonomi-filogenetik.html#cb11-7" aria-hidden="true" tabindex="-1"></a><span class="co"># BiocManager::install(&quot;DECIPHER&quot;)</span></span>
<span id="cb11-8"><a href="taksonomi-filogenetik.html#cb11-8" aria-hidden="true" tabindex="-1"></a><span class="co"># install.packages(&quot;ape&quot;)</span></span>
<span id="cb11-9"><a href="taksonomi-filogenetik.html#cb11-9" aria-hidden="true" tabindex="-1"></a><span class="co"># install.packages(&quot;plotly&quot;)</span></span>
<span id="cb11-10"><a href="taksonomi-filogenetik.html#cb11-10" aria-hidden="true" tabindex="-1"></a><span class="co"># install.packages(&quot;phytools&quot;)</span></span></code></pre></div>
<p>Setelah proses instalasi selesai, anda perlu melakukan pemanggilan packages untuk digunakan dengan menjalankan code di bawah ini.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="taksonomi-filogenetik.html#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(seqinr)</span></code></pre></div>
<pre><code>## 
## Attaching package: &#39;seqinr&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:dplyr&#39;:
## 
##     count</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="taksonomi-filogenetik.html#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(adegenet)</span></code></pre></div>
<pre><code>## Loading required package: ade4</code></pre>
<pre><code>## 
##    /// adegenet 2.1.5 is loaded ////////////
## 
##    &gt; overview: &#39;?adegenet&#39;
##    &gt; tutorials/doc/questions: &#39;adegenetWeb()&#39; 
##    &gt; bug reports/feature requests: adegenetIssues()</code></pre>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="taksonomi-filogenetik.html#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ape)</span></code></pre></div>
<pre><code>## 
## Attaching package: &#39;ape&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:seqinr&#39;:
## 
##     as.alignment, consensus</code></pre>
<pre><code>## The following object is masked from &#39;package:flextable&#39;:
## 
##     rotate</code></pre>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="taksonomi-filogenetik.html#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggtree)</span></code></pre></div>
<pre><code>## Registered S3 method overwritten by &#39;ggtree&#39;:
##   method      from 
##   identify.gg ggfun</code></pre>
<pre><code>## ggtree v3.2.1  For help: https://yulab-smu.top/treedata-book/
## 
## If you use ggtree in published research, please cite the most appropriate paper(s):
## 
## 1. Guangchuang Yu. Using ggtree to visualize data on tree-like structures. Current Protocols in Bioinformatics. 2020, 69:e96. doi:10.1002/cpbi.96
## 2. Guangchuang Yu, Tommy Tsan-Yuk Lam, Huachen Zhu, Yi Guan. Two methods for mapping and visualizing associated data on phylogeny using ggtree. Molecular Biology and Evolution. 2018, 35(12):3041-3043. doi:10.1093/molbev/msy194
## 3. Guangchuang Yu, David Smith, Huachen Zhu, Yi Guan, Tommy Tsan-Yuk Lam. ggtree: an R package for visualization and annotation of phylogenetic trees with their covariates and other associated data. Methods in Ecology and Evolution. 2017, 8(1):28-36. doi:10.1111/2041-210X.12628</code></pre>
<pre><code>## 
## Attaching package: &#39;ggtree&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:ape&#39;:
## 
##     rotate</code></pre>
<pre><code>## The following object is masked from &#39;package:tidyr&#39;:
## 
##     expand</code></pre>
<pre><code>## The following object is masked from &#39;package:flextable&#39;:
## 
##     rotate</code></pre>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="taksonomi-filogenetik.html#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DECIPHER)</span></code></pre></div>
<pre><code>## Loading required package: Biostrings</code></pre>
<pre><code>## Loading required package: BiocGenerics</code></pre>
<pre><code>## 
## Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:ade4&#39;:
## 
##     score</code></pre>
<pre><code>## The following objects are masked from &#39;package:dplyr&#39;:
## 
##     combine, intersect, setdiff, union</code></pre>
<pre><code>## The following object is masked from &#39;package:flextable&#39;:
## 
##     width</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     IQR, mad, sd, var, xtabs</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     anyDuplicated, append, as.data.frame, basename, cbind, colnames,
##     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
##     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
##     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
##     union, unique, unsplit, which.max, which.min</code></pre>
<pre><code>## Loading required package: S4Vectors</code></pre>
<pre><code>## Loading required package: stats4</code></pre>
<pre><code>## 
## Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:ggtree&#39;:
## 
##     expand</code></pre>
<pre><code>## The following objects are masked from &#39;package:dplyr&#39;:
## 
##     first, rename</code></pre>
<pre><code>## The following object is masked from &#39;package:tidyr&#39;:
## 
##     expand</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     expand.grid, I, unname</code></pre>
<pre><code>## Loading required package: IRanges</code></pre>
<pre><code>## 
## Attaching package: &#39;IRanges&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:ggtree&#39;:
## 
##     collapse</code></pre>
<pre><code>## The following objects are masked from &#39;package:dplyr&#39;:
## 
##     collapse, desc, slice</code></pre>
<pre><code>## The following object is masked from &#39;package:purrr&#39;:
## 
##     reduce</code></pre>
<pre><code>## The following object is masked from &#39;package:grDevices&#39;:
## 
##     windows</code></pre>
<pre><code>## Loading required package: XVector</code></pre>
<pre><code>## 
## Attaching package: &#39;XVector&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:purrr&#39;:
## 
##     compact</code></pre>
<pre><code>## Loading required package: GenomeInfoDb</code></pre>
<pre><code>## 
## Attaching package: &#39;Biostrings&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:ape&#39;:
## 
##     complement</code></pre>
<pre><code>## The following object is masked from &#39;package:seqinr&#39;:
## 
##     translate</code></pre>
<pre><code>## The following object is masked from &#39;package:base&#39;:
## 
##     strsplit</code></pre>
<pre><code>## Loading required package: RSQLite</code></pre>
<pre><code>## Loading required package: parallel</code></pre>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="taksonomi-filogenetik.html#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(viridis)</span></code></pre></div>
<pre><code>## Loading required package: viridisLite</code></pre>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="taksonomi-filogenetik.html#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb63-2"><a href="taksonomi-filogenetik.html#cb63-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(here)</span></code></pre></div>
<pre><code>## here() starts at C:/Users/Lenovo/Documents/praktikumsismikbits</code></pre>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="taksonomi-filogenetik.html#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(plotly)</span></code></pre></div>
<pre><code>## 
## Attaching package: &#39;plotly&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:XVector&#39;:
## 
##     slice</code></pre>
<pre><code>## The following object is masked from &#39;package:IRanges&#39;:
## 
##     slice</code></pre>
<pre><code>## The following object is masked from &#39;package:S4Vectors&#39;:
## 
##     rename</code></pre>
<pre><code>## The following object is masked from &#39;package:ggplot2&#39;:
## 
##     last_plot</code></pre>
<pre><code>## The following objects are masked from &#39;package:flextable&#39;:
## 
##     highlight, style</code></pre>
<pre><code>## The following object is masked from &#39;package:stats&#39;:
## 
##     filter</code></pre>
<pre><code>## The following object is masked from &#39;package:graphics&#39;:
## 
##     layout</code></pre>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="taksonomi-filogenetik.html#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phytools)</span></code></pre></div>
<pre><code>## Loading required package: maps</code></pre>
<pre><code>## 
## Attaching package: &#39;maps&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:viridis&#39;:
## 
##     unemp</code></pre>
<pre><code>## The following object is masked from &#39;package:purrr&#39;:
## 
##     map</code></pre>
<p>Untuk melakukan pengecekan file path atau directory, gunakan package “here”. Kemudian buatlah folder khusus pada directory yang muncul. Buatlah folder “data”, “output”</p>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="taksonomi-filogenetik.html#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="fu">here</span>()</span>
<span id="cb79-2"><a href="taksonomi-filogenetik.html#cb79-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="st">&quot;C:/Users/Lenovo/Documents/ggtree advance/data&quot;</span>)</span></code></pre></div>
<p>Pada tahap awal, anda perlu melakukan pemanggilan data ke dalam R dengan menjalankan code di bawah ini</p>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="taksonomi-filogenetik.html#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="co"># masukkan atau impor data sequence dari drive ke R</span></span>
<span id="cb80-2"><a href="taksonomi-filogenetik.html#cb80-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Ubah &quot;DNA&quot; ke &quot;RNA&quot; atau &quot;AA&quot; jika diperlukan </span></span>
<span id="cb80-3"><a href="taksonomi-filogenetik.html#cb80-3" aria-hidden="true" tabindex="-1"></a>seqs <span class="ot">&lt;-</span> <span class="fu">readDNAStringSet</span>(<span class="st">&quot;C:/Users/Lenovo/Documents/praktikumsismikbits/data/bacillus.txt&quot;</span>, <span class="at">format =</span> <span class="st">&quot;fasta&quot;</span>)</span></code></pre></div>
<p>Kita bisa melihat sequence sekilas</p>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="taksonomi-filogenetik.html#cb81-1" aria-hidden="true" tabindex="-1"></a>seqs</span></code></pre></div>
<pre><code>## DNAStringSet object of length 10:
##      width seq                                              names               
##  [1]  1427 GAGAGTTTGATCCTGGCTCAGGA...ACCACGAGAGTTTGTAACACCC NR_118950.1 Bacil...
##  [2]  1396 CGAGCGGACTTAGCGGGAGCTTG...ACACCCGAAGTCGGTGGGGTAA NR_171460.1 Bacil...
##  [3]  1458 GCACGTGCGGCGTGCTATACATG...GCCTAAGTGGGACAGATAGGGG NR_170510.1 Bacil...
##  [4]  1576 TTTTAGATGCTAGACAAACTAAC...GTGCGGCTGGATCACCTCCTTT NR_170494.1 Bacil...
##  [5]  1503 AGTTTGATCCTGGCTCAGGACGA...ATGATTGGGGTGAAGTCGTAAC NR_170424.1 Bacil...
##  [6]  1476 GACGAACGCTGGCGGCGTGCCTA...GGTGGGACAGGTGATTGGGGTG NR_165783.1 Bacil...
##  [7]  1396 CGAACGCTGGCGGCGTGCCTAAT...TGTACACACCGCCCGTCACACC NR_165694.1 Bacil...
##  [8]  1474 CAACCGCGGCCGCCGTCCTTATT...GGACAGATGATTGGGGTGAAGT NR_165685.1 Bacil...
##  [9]  1438 GGCTATCATGCAAGTCGAGCGAA...GGAGCAGCCGCCAAGTGACGTG NR_164882.1 Bacil...
## [10]  1426 TGCAGTCGAGCGACGGATGGAGA...TTACGGGAGCCAGCCGCCTAAG NR_159907.1 Bacil...</code></pre>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="taksonomi-filogenetik.html#cb83-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Sebagai catatan, sekuens harus memiliki orientasi yang sama</span></span>
<span id="cb83-2"><a href="taksonomi-filogenetik.html#cb83-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Jika tidak, sequence dapat di- reorientasi (optional)</span></span>
<span id="cb83-3"><a href="taksonomi-filogenetik.html#cb83-3" aria-hidden="true" tabindex="-1"></a>seqs <span class="ot">&lt;-</span> <span class="fu">OrientNucleotides</span>(seqs)</span></code></pre></div>
<pre><code>## ========================================================================================================================================================================================================
## 
## Time difference of 0.15 secs</code></pre>
<p>Dalam membentuk pohon filogenetik, kita perlu melakukan alignment terhadap sequence terlebih dahulu</p>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="taksonomi-filogenetik.html#cb85-1" aria-hidden="true" tabindex="-1"></a>aligned <span class="ot">&lt;-</span> <span class="fu">AlignSeqs</span>(seqs)</span></code></pre></div>
<pre><code>## Determining distance matrix based on shared 9-mers:
## ================================================================================
## 
## Time difference of 0.02 secs
## 
## Clustering into groups by similarity:
## ================================================================================
## 
## Time difference of 0.02 secs
## 
## Aligning Sequences:
## ================================================================================
## 
## Time difference of 0.31 secs
## 
## Iteration 1 of 2:
## 
## Determining distance matrix based on alignment:
## ================================================================================
## 
## Time difference of 0.02 secs
## 
## Reclustering into groups by similarity:
## ================================================================================
## 
## Time difference of 0 secs
## 
## Realigning Sequences:
## ================================================================================
## 
## Time difference of 0.22 secs
## 
## Iteration 2 of 2:
## 
## Determining distance matrix based on alignment:
## ================================================================================
## 
## Time difference of 0 secs
## 
## Reclustering into groups by similarity:
## ================================================================================
## 
## Time difference of 0 secs
## 
## Realigning Sequences:
## ================================================================================
## 
## Time difference of 0.02 secs</code></pre>
<p>untuk melihat hasil alignment, jalankan code berikut:</p>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="taksonomi-filogenetik.html#cb87-1" aria-hidden="true" tabindex="-1"></a><span class="fu">BrowseSeqs</span>(aligned, <span class="at">highlight=</span><span class="dv">0</span>)</span></code></pre></div>
<p>Setelah sequence aligned, kita gabungkan hasil tadi menjadi 1 fasta file</p>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="taksonomi-filogenetik.html#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="fu">writeXStringSet</span>(aligned,</span>
<span id="cb88-2"><a href="taksonomi-filogenetik.html#cb88-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">file=</span><span class="st">&quot;bacillus_aligned.fasta&quot;</span>)</span>
<span id="cb88-3"><a href="taksonomi-filogenetik.html#cb88-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-4"><a href="taksonomi-filogenetik.html#cb88-4" aria-hidden="true" tabindex="-1"></a><span class="co"># kemudian, read sequenced yang telah di-align tadi</span></span>
<span id="cb88-5"><a href="taksonomi-filogenetik.html#cb88-5" aria-hidden="true" tabindex="-1"></a>dna <span class="ot">&lt;-</span> <span class="fu">read.alignment</span>(<span class="st">&quot;bacillus_aligned.fasta&quot;</span>, <span class="at">format =</span> <span class="st">&quot;fasta&quot;</span>)</span></code></pre></div>
<p>Kemudian, dilakukan perhitungan distance matrix</p>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="taksonomi-filogenetik.html#cb89-1" aria-hidden="true" tabindex="-1"></a><span class="co"># buat distance matrix untuk file alignment </span></span>
<span id="cb89-2"><a href="taksonomi-filogenetik.html#cb89-2" aria-hidden="true" tabindex="-1"></a>D <span class="ot">&lt;-</span> <span class="fu">dist.alignment</span>(dna, <span class="at">matrix =</span> <span class="st">&quot;similarity&quot;</span>)</span>
<span id="cb89-3"><a href="taksonomi-filogenetik.html#cb89-3" aria-hidden="true" tabindex="-1"></a>temp <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">as.matrix</span>(D))</span>
<span id="cb89-4"><a href="taksonomi-filogenetik.html#cb89-4" aria-hidden="true" tabindex="-1"></a><span class="fu">table.paint</span>(temp, <span class="at">cleg=</span><span class="dv">0</span>, <span class="at">clabel.row=</span>.<span class="dv">5</span>, <span class="at">clabel.col=</span>.<span class="dv">5</span>)<span class="sc">+</span></span>
<span id="cb89-5"><a href="taksonomi-filogenetik.html#cb89-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_viridis</span>()<span class="co">#darker shades of gray = distance yang lebih besar </span></span></code></pre></div>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<pre><code>## NULL</code></pre>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="taksonomi-filogenetik.html#cb91-1" aria-hidden="true" tabindex="-1"></a><span class="co"># kita bisa melihat pola karena data disusun berdasarkan tahun </span></span>
<span id="cb91-2"><a href="taksonomi-filogenetik.html#cb91-2" aria-hidden="true" tabindex="-1"></a><span class="co"># namun, kita belum bisa mencapai ke kesimpulan</span></span></code></pre></div>
<p>Setelah distance matrix telah terbentuk, proses pembuatan pohon filogenetik dimulai.</p>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="taksonomi-filogenetik.html#cb92-1" aria-hidden="true" tabindex="-1"></a>tre <span class="ot">&lt;-</span> <span class="fu">nj</span>(D)</span>
<span id="cb92-2"><a href="taksonomi-filogenetik.html#cb92-2" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(tre) <span class="co">#semua tree yang divuat dengan {ape} package akan termasuk class phylo</span></span></code></pre></div>
<pre><code>## [1] &quot;phylo&quot;</code></pre>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="taksonomi-filogenetik.html#cb94-1" aria-hidden="true" tabindex="-1"></a>tre <span class="ot">&lt;-</span> <span class="fu">ladderize</span>(tre)</span>
<span id="cb94-2"><a href="taksonomi-filogenetik.html#cb94-2" aria-hidden="true" tabindex="-1"></a><span class="co"># pada tahap ini, pohon filogenetik telah terbentuk, setelah ini dilakukan proses plotting</span></span></code></pre></div>
<p>Tahap memplotting di sini dapat menggunakan 2 cara, yaitu:
1. Base R
2. ggtree package</p>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="taksonomi-filogenetik.html#cb95-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Menggunakan base R</span></span>
<span id="cb95-2"><a href="taksonomi-filogenetik.html#cb95-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(tre, <span class="at">cex =</span> <span class="fl">0.6</span>)</span>
<span id="cb95-3"><a href="taksonomi-filogenetik.html#cb95-3" aria-hidden="true" tabindex="-1"></a><span class="fu">title</span>(<span class="st">&quot;Bacillus&quot;</span>)</span></code></pre></div>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-14-1.png" width="672" /></p>
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="taksonomi-filogenetik.html#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="co"># atau</span></span>
<span id="cb96-2"><a href="taksonomi-filogenetik.html#cb96-2" aria-hidden="true" tabindex="-1"></a>h_cluster <span class="ot">&lt;-</span> <span class="fu">hclust</span>(D, <span class="at">method =</span> <span class="st">&quot;average&quot;</span>, <span class="at">members =</span> <span class="cn">NULL</span>) <span class="co"># method = average is used for UPGMA, members can be equal to NULL or a vector with a length of size D</span></span>
<span id="cb96-3"><a href="taksonomi-filogenetik.html#cb96-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(h_cluster, <span class="at">cex =</span> <span class="fl">0.6</span>)</span></code></pre></div>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-14-2.png" width="672" /></p>
<p>Penggunaan ggtree sangat fleksible. Anda dapat melakukan kustomisasi pohon filogenetik dengan mudah dan cepat.</p>
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="taksonomi-filogenetik.html#cb97-1" aria-hidden="true" tabindex="-1"></a><span class="co"># alaterntaif pertama</span></span>
<span id="cb97-2"><a href="taksonomi-filogenetik.html#cb97-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tre, <span class="at">yscale =</span> <span class="st">&quot;NA&quot;</span>)<span class="sc">+</span></span>
<span id="cb97-3"><a href="taksonomi-filogenetik.html#cb97-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tiplab</span>(<span class="at">hjust =</span> <span class="sc">-</span><span class="fl">0.3</span>, <span class="at">size=</span><span class="dv">4</span>, <span class="at">align =</span> <span class="cn">TRUE</span>)<span class="sc">+</span></span>
<span id="cb97-4"><a href="taksonomi-filogenetik.html#cb97-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlim</span>(<span class="dv">0</span>,<span class="fl">0.5</span>) </span></code></pre></div>
<pre><code>## Warning in scaleY(as.phylo(model), res, yscale, layout, ...): yscale is not available...</code></pre>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-15-1.png" width="672" /></p>
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb99-1"><a href="taksonomi-filogenetik.html#cb99-1" aria-hidden="true" tabindex="-1"></a><span class="co"># cara lain</span></span>
<span id="cb99-2"><a href="taksonomi-filogenetik.html#cb99-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tre,<span class="at">layout =</span> <span class="st">&quot;daylight&quot;</span>)<span class="sc">+</span></span>
<span id="cb99-3"><a href="taksonomi-filogenetik.html#cb99-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tiplab</span>(<span class="at">hjust =</span> <span class="sc">-</span><span class="fl">0.3</span>, <span class="at">size=</span><span class="dv">4</span>, <span class="at">align =</span> <span class="cn">TRUE</span>)<span class="sc">+</span></span>
<span id="cb99-4"><a href="taksonomi-filogenetik.html#cb99-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlim</span>(<span class="dv">0</span>,<span class="fl">0.5</span>) </span></code></pre></div>
<pre><code>## Average angle change [1] 0.119040575095853</code></pre>
<pre><code>## Average angle change [2] 0.0180065062992894</code></pre>
<pre><code>## Warning: Removed 5 rows containing missing values (geom_segment_g_gtree).</code></pre>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-15-2.png" width="672" /></p>
<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb103-1"><a href="taksonomi-filogenetik.html#cb103-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot a basic tree</span></span>
<span id="cb103-2"><a href="taksonomi-filogenetik.html#cb103-2" aria-hidden="true" tabindex="-1"></a>treeplot <span class="ot">&lt;-</span> <span class="fu">ggtree</span>(tre) <span class="sc">+</span> </span>
<span id="cb103-3"><a href="taksonomi-filogenetik.html#cb103-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tiplab</span>(<span class="at">hjust =</span> <span class="sc">-</span><span class="fl">0.3</span>, <span class="at">size=</span><span class="dv">4</span>, <span class="at">align =</span> <span class="cn">TRUE</span>)<span class="sc">+</span></span>
<span id="cb103-4"><a href="taksonomi-filogenetik.html#cb103-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlim</span>(<span class="dv">0</span>,<span class="fl">0.5</span>)</span>
<span id="cb103-5"><a href="taksonomi-filogenetik.html#cb103-5" aria-hidden="true" tabindex="-1"></a>treeplot</span></code></pre></div>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-15-3.png" width="672" /></p>
<p>Terdapat beberapa bentuk pohon filogenetik yang dapat anda gunakan yaitu:
1. Rectangular
2. Circular
3. Slanted</p>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="taksonomi-filogenetik.html#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tre,</span>
<span id="cb104-2"><a href="taksonomi-filogenetik.html#cb104-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">layout =</span> <span class="st">&quot;rectangular&quot;</span>) <span class="co"># did anything change? this is the default</span></span></code></pre></div>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-16-1.png" width="672" /></p>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="taksonomi-filogenetik.html#cb105-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tre, </span>
<span id="cb105-2"><a href="taksonomi-filogenetik.html#cb105-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">layout =</span> <span class="st">&quot;circular&quot;</span>) </span></code></pre></div>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-16-2.png" width="672" /></p>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="taksonomi-filogenetik.html#cb106-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tre, </span>
<span id="cb106-2"><a href="taksonomi-filogenetik.html#cb106-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">layout =</span> <span class="st">&quot;slanted&quot;</span>)</span></code></pre></div>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-16-3.png" width="672" /></p>
<p>Anda juga dapat membentuk kladogram dengan mengatur panjang branch</p>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="taksonomi-filogenetik.html#cb107-1" aria-hidden="true" tabindex="-1"></a><span class="co"># mengatur branch.length = &quot;none&quot; akan membentuk cladogram</span></span>
<span id="cb107-2"><a href="taksonomi-filogenetik.html#cb107-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tre,</span>
<span id="cb107-3"><a href="taksonomi-filogenetik.html#cb107-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">layout =</span> <span class="st">&quot;rectangular&quot;</span>, <span class="at">branch.length =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-17-1.png" width="672" /></p>
<p>Anda juga dapat berinteraksi dengan plot dengan plotly package</p>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="taksonomi-filogenetik.html#cb108-1" aria-hidden="true" tabindex="-1"></a><span class="co"># melihat plot dengan interaktif</span></span>
<span id="cb108-2"><a href="taksonomi-filogenetik.html#cb108-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplotly</span>(treeplot)</span></code></pre></div>
<pre><code>## Warning in geom2trace.default(dots[[1L]][[1L]], dots[[2L]][[1L]], dots[[3L]][[1L]]): geom_GeomSegmentGGtree() has yet to be implemented in plotly.
##   If you&#39;d like to see this geom implemented,
##   Please open an issue with your example code at
##   https://github.com/ropensci/plotly/issues</code></pre>
<pre><code>## Warning in geom2trace.default(dots[[1L]][[1L]], dots[[2L]][[1L]], dots[[3L]][[1L]]): geom_GeomTextGGtree() has yet to be implemented in plotly.
##   If you&#39;d like to see this geom implemented,
##   Please open an issue with your example code at
##   https://github.com/ropensci/plotly/issues</code></pre>
<div id="htmlwidget-fadaad138a18436b8a88" style="width:672px;height:480px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-fadaad138a18436b8a88">{"x":{"data":[{"x":[0.0333880320802013,0.121177864218752,null,0.0189625250097535,0.129790553983827,null,0.0312711433274793,0.131715733322054,null,0.0103911092628414,0.138105801842833,null,0,0.119791778334951,null,0.00886431540207036,0.166400473254252,null,0.0103911092628414,0.119392454919849,null,0.0333880320802013,0.168559666482305,null,0.0312711433274793,0.143655515759043,null,0.0113737111812037,0.118366184280933,null,0,0,null,0,0.00435148761414486,null,0.00435148761414486,0.0103911092628414,null,0.00435148761414486,0.0113737111812037,null,0,0.00886431540207036,null,0.0113737111812037,0.0189625250097535,null,0.0189625250097535,0.0312711433274793,null,0.00886431540207036,0.0333880320802013],"y":[3,3,null,8,8,null,9,9,null,5,5,null,1,1,null,2,2,null,6,6,null,4,4,null,10,10,null,7,7,null,3.47916666666667,3.47916666666667,null,6.6875,6.6875,null,5.5,5.5,null,7.875,7.875,null,2.75,2.75,null,8.75,8.75,null,9.5,9.5,null,3.5,3.5],"text":["x: 0.121177864<br />y:  3.000000<br />node:  1<br />parent: 18","x: 0.121177864<br />y:  3.000000<br />node:  1<br />parent: 18",null,"x: 0.129790554<br />y:  8.000000<br />node:  2<br />parent: 16","x: 0.129790554<br />y:  8.000000<br />node:  2<br />parent: 16",null,"x: 0.131715733<br />y:  9.000000<br />node:  3<br />parent: 17","x: 0.131715733<br />y:  9.000000<br />node:  3<br />parent: 17",null,"x: 0.138105802<br />y:  5.000000<br />node:  4<br />parent: 13","x: 0.138105802<br />y:  5.000000<br />node:  4<br />parent: 13",null,"x: 0.119791778<br />y:  1.000000<br />node:  5<br />parent: 11","x: 0.119791778<br />y:  1.000000<br />node:  5<br />parent: 11",null,"x: 0.166400473<br />y:  2.000000<br />node:  6<br />parent: 15","x: 0.166400473<br />y:  2.000000<br />node:  6<br />parent: 15",null,"x: 0.119392455<br />y:  6.000000<br />node:  7<br />parent: 13","x: 0.119392455<br />y:  6.000000<br />node:  7<br />parent: 13",null,"x: 0.168559666<br />y:  4.000000<br />node:  8<br />parent: 18","x: 0.168559666<br />y:  4.000000<br />node:  8<br />parent: 18",null,"x: 0.143655516<br />y: 10.000000<br />node:  9<br />parent: 17","x: 0.143655516<br />y: 10.000000<br />node:  9<br />parent: 17",null,"x: 0.118366184<br />y:  7.000000<br />node: 10<br />parent: 14","x: 0.118366184<br />y:  7.000000<br />node: 10<br />parent: 14",null,"x: 0.000000000<br />y:  3.479167<br />node: 11<br />parent: 11","x: 0.000000000<br />y:  3.479167<br />node: 11<br />parent: 11",null,"x: 0.004351488<br />y:  6.687500<br />node: 12<br />parent: 11","x: 0.004351488<br />y:  6.687500<br />node: 12<br />parent: 11",null,"x: 0.010391109<br />y:  5.500000<br />node: 13<br />parent: 12","x: 0.010391109<br />y:  5.500000<br />node: 13<br />parent: 12",null,"x: 0.011373711<br />y:  7.875000<br />node: 14<br />parent: 12","x: 0.011373711<br />y:  7.875000<br />node: 14<br />parent: 12",null,"x: 0.008864315<br />y:  2.750000<br />node: 15<br />parent: 11","x: 0.008864315<br />y:  2.750000<br />node: 15<br />parent: 11",null,"x: 0.018962525<br />y:  8.750000<br />node: 16<br />parent: 14","x: 0.018962525<br />y:  8.750000<br />node: 16<br />parent: 14",null,"x: 0.031271143<br />y:  9.500000<br />node: 17<br />parent: 16","x: 0.031271143<br />y:  9.500000<br />node: 17<br />parent: 16",null,"x: 0.033388032<br />y:  3.500000<br />node: 18<br />parent: 15","x: 0.033388032<br />y:  3.500000<br />node: 18<br />parent: 15"],"type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.0333880320802013,0.0333880320802013,null,0.0189625250097535,0.0189625250097535,null,0.0312711433274793,0.0312711433274793,null,0.0103911092628414,0.0103911092628414,null,0,0,null,0.00886431540207036,0.00886431540207036,null,0.0103911092628414,0.0103911092628414,null,0.0333880320802013,0.0333880320802013,null,0.0312711433274793,0.0312711433274793,null,0.0113737111812037,0.0113737111812037,null,0,0,null,0,0,null,0.00435148761414486,0.00435148761414486,null,0.00435148761414486,0.00435148761414486,null,0,0,null,0.0113737111812037,0.0113737111812037,null,0.0189625250097535,0.0189625250097535,null,0.00886431540207036,0.00886431540207036],"y":[3.5,3,null,8.75,8,null,9.5,9,null,5.5,5,null,3.47916666666667,1,null,2.75,2,null,5.5,6,null,3.5,4,null,9.5,10,null,7.875,7,null,3.47916666666667,3.47916666666667,null,3.47916666666667,6.6875,null,6.6875,5.5,null,6.6875,7.875,null,3.47916666666667,2.75,null,7.875,8.75,null,8.75,9.5,null,2.75,3.5],"text":["x: 0.121177864<br />y:  3.000000<br />node:  1<br />parent: 18","x: 0.121177864<br />y:  3.000000<br />node:  1<br />parent: 18",null,"x: 0.129790554<br />y:  8.000000<br />node:  2<br />parent: 16","x: 0.129790554<br />y:  8.000000<br />node:  2<br />parent: 16",null,"x: 0.131715733<br />y:  9.000000<br />node:  3<br />parent: 17","x: 0.131715733<br />y:  9.000000<br />node:  3<br />parent: 17",null,"x: 0.138105802<br />y:  5.000000<br />node:  4<br />parent: 13","x: 0.138105802<br />y:  5.000000<br />node:  4<br />parent: 13",null,"x: 0.119791778<br />y:  1.000000<br />node:  5<br />parent: 11","x: 0.119791778<br />y:  1.000000<br />node:  5<br />parent: 11",null,"x: 0.166400473<br />y:  2.000000<br />node:  6<br />parent: 15","x: 0.166400473<br />y:  2.000000<br />node:  6<br />parent: 15",null,"x: 0.119392455<br />y:  6.000000<br />node:  7<br />parent: 13","x: 0.119392455<br />y:  6.000000<br />node:  7<br />parent: 13",null,"x: 0.168559666<br />y:  4.000000<br />node:  8<br />parent: 18","x: 0.168559666<br />y:  4.000000<br />node:  8<br />parent: 18",null,"x: 0.143655516<br />y: 10.000000<br />node:  9<br />parent: 17","x: 0.143655516<br />y: 10.000000<br />node:  9<br />parent: 17",null,"x: 0.118366184<br />y:  7.000000<br />node: 10<br />parent: 14","x: 0.118366184<br />y:  7.000000<br />node: 10<br />parent: 14",null,"x: 0.000000000<br />y:  3.479167<br />node: 11<br />parent: 11","x: 0.000000000<br />y:  3.479167<br />node: 11<br />parent: 11",null,"x: 0.004351488<br />y:  6.687500<br />node: 12<br />parent: 11","x: 0.004351488<br />y:  6.687500<br />node: 12<br />parent: 11",null,"x: 0.010391109<br />y:  5.500000<br />node: 13<br />parent: 12","x: 0.010391109<br />y:  5.500000<br />node: 13<br />parent: 12",null,"x: 0.011373711<br />y:  7.875000<br />node: 14<br />parent: 12","x: 0.011373711<br />y:  7.875000<br />node: 14<br />parent: 12",null,"x: 0.008864315<br />y:  2.750000<br />node: 15<br />parent: 11","x: 0.008864315<br />y:  2.750000<br />node: 15<br />parent: 11",null,"x: 0.018962525<br />y:  8.750000<br />node: 16<br />parent: 14","x: 0.018962525<br />y:  8.750000<br />node: 16<br />parent: 14",null,"x: 0.031271143<br />y:  9.500000<br />node: 17<br />parent: 16","x: 0.031271143<br />y:  9.500000<br />node: 17<br />parent: 16",null,"x: 0.033388032<br />y:  3.500000<br />node: 18<br />parent: 15","x: 0.033388032<br />y:  3.500000<br />node: 18<br />parent: 15"],"type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","mode":"","frame":null}],"layout":{"margin":{"t":26.2283105022831,"r":7.30593607305936,"b":13.8812785388128,"l":10.958904109589},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-0.025,0.525],"tickmode":"array","ticktext":["0.0","0.1","0.2","0.3","0.4","0.5"],"tickvals":[0,0.1,0.2,0.3,0.4,0.5],"categoryorder":"array","categoryarray":["0.0","0.1","0.2","0.3","0.4","0.5"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.65296803652968,"tickwidth":0,"showticklabels":false,"tickfont":{"color":null,"family":null,"size":0},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":{"text":"","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187}},"hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0.4,10.6],"tickmode":"array","ticktext":["2.5","5.0","7.5","10.0"],"tickvals":[2.5,5,7.5,10],"categoryorder":"array","categoryarray":["2.5","5.0","7.5","10.0"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.65296803652968,"tickwidth":0,"showticklabels":false,"tickfont":{"color":null,"family":null,"size":0},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":{"text":"","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187}},"hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1}],"showlegend":false,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895}},"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"source":"A","attrs":{"20c47adc58e":{"x":{},"y":{},"node":{},"parent":{},"type":"scatter"},"20c4439d7465":{"x":{},"y":{},"node":{},"parent":{}},"20c424c79ed":{"x":{},"y":{},"xend":{},"yend":{},"node":{},"label":{},"subset":{}},"20c45d572dc5":{"x":{},"y":{},"label":{},"node":{},"subset":{}}},"cur_data":"20c47adc58e","visdat":{"20c47adc58e":["function (y) ","x"],"20c4439d7465":["function (y) ","x"],"20c424c79ed":["function (y) ","x"],"20c45d572dc5":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
<div id="kustomisasi-lebih-lanjut" class="section level6 hasAnchor" number="2.1.3.3.0.1">
<h6><span class="header-section-number">2.1.3.3.0.1</span> Kustomisasi lebih lanjut<a href="taksonomi-filogenetik.html#kustomisasi-lebih-lanjut" class="anchor-section" aria-label="Anchor link to header"></a></h6>
<div class="sourceCode" id="cb111"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb111-1"><a href="taksonomi-filogenetik.html#cb111-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot using ggtree and highlight clusters</span></span>
<span id="cb111-2"><a href="taksonomi-filogenetik.html#cb111-2" aria-hidden="true" tabindex="-1"></a><span class="co"># change the node values for your own data</span></span>
<span id="cb111-3"><a href="taksonomi-filogenetik.html#cb111-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tre) <span class="sc">+</span> </span>
<span id="cb111-4"><a href="taksonomi-filogenetik.html#cb111-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tiplab</span>(<span class="at">hjust =</span> <span class="sc">-</span><span class="fl">0.3</span>, <span class="at">size=</span><span class="dv">4</span>, <span class="at">align =</span> <span class="cn">TRUE</span>) <span class="sc">+</span> </span>
<span id="cb111-5"><a href="taksonomi-filogenetik.html#cb111-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hilight</span>(<span class="at">node=</span><span class="dv">5</span>, <span class="at">fill=</span><span class="st">&quot;purple&quot;</span>, <span class="at">alpha =</span> <span class="fl">0.2</span>) <span class="sc">+</span> </span>
<span id="cb111-6"><a href="taksonomi-filogenetik.html#cb111-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hilight</span>(<span class="at">node=</span><span class="dv">15</span>, <span class="at">fill=</span><span class="st">&quot;dark green&quot;</span>, <span class="at">alpha =</span> <span class="fl">0.2</span>) <span class="sc">+</span></span>
<span id="cb111-7"><a href="taksonomi-filogenetik.html#cb111-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hilight</span>(<span class="at">node=</span><span class="dv">12</span>, <span class="at">fill=</span><span class="st">&quot;gold&quot;</span>, <span class="at">alpha =</span> <span class="fl">0.2</span>) <span class="sc">+</span></span>
<span id="cb111-8"><a href="taksonomi-filogenetik.html#cb111-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlim</span>(<span class="dv">0</span>,<span class="fl">0.5</span>) </span></code></pre></div>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-19-1.png" width="672" />
Menambahkan label di pohon filogenetik</p>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="taksonomi-filogenetik.html#cb112-1" aria-hidden="true" tabindex="-1"></a><span class="co">#tiplab</span></span>
<span id="cb112-2"><a href="taksonomi-filogenetik.html#cb112-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(<span class="fu">midpoint.root</span>(tre)) <span class="sc">+</span> </span>
<span id="cb112-3"><a href="taksonomi-filogenetik.html#cb112-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_treescale</span>(<span class="at">x =</span> <span class="dv">0</span>, <span class="at">y =</span> <span class="dv">0</span>, <span class="co"># x and y position of the treescale</span></span>
<span id="cb112-4"><a href="taksonomi-filogenetik.html#cb112-4" aria-hidden="true" tabindex="-1"></a>                 <span class="at">width =</span> <span class="fl">0.01</span>) <span class="sc">+</span> <span class="co"># width of scale</span></span>
<span id="cb112-5"><a href="taksonomi-filogenetik.html#cb112-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tiplab</span>(<span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span> <span class="co"># displaying tip labels </span></span>
<span id="cb112-6"><a href="taksonomi-filogenetik.html#cb112-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">clip =</span> <span class="st">&#39;off&#39;</span>)<span class="sc">+</span> <span class="co"># allows us to draw outside the plot</span></span>
<span id="cb112-7"><a href="taksonomi-filogenetik.html#cb112-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.margin =</span> <span class="fu">margin</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">1</span>, <span class="st">&quot;cm&quot;</span>)) <span class="co"># add space around the plot</span></span></code></pre></div>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-20-1.png" width="672" /></p>
<p>Menambahkan keterangan pada node</p>
<div class="sourceCode" id="cb113"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb113-1"><a href="taksonomi-filogenetik.html#cb113-1" aria-hidden="true" tabindex="-1"></a><span class="co"># change the node values for your own data</span></span>
<span id="cb113-2"><a href="taksonomi-filogenetik.html#cb113-2" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">ggtree</span>(tre) <span class="sc">+</span> </span>
<span id="cb113-3"><a href="taksonomi-filogenetik.html#cb113-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tiplab</span>( <span class="at">size=</span><span class="dv">3</span>, <span class="at">align =</span> <span class="cn">TRUE</span>) <span class="sc">+</span> </span>
<span id="cb113-4"><a href="taksonomi-filogenetik.html#cb113-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_nodepoint</span>(<span class="at">color =</span> <span class="st">&quot;orange&quot;</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>)</span></code></pre></div>
<p>Melakukan plotting sequence dengan pohon filogenetik</p>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="taksonomi-filogenetik.html#cb114-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot the alignment </span></span>
<span id="cb114-2"><a href="taksonomi-filogenetik.html#cb114-2" aria-hidden="true" tabindex="-1"></a>dnanih <span class="ot">&lt;-</span> <span class="fu">as.DNAbin</span>(dna)</span>
<span id="cb114-3"><a href="taksonomi-filogenetik.html#cb114-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-4"><a href="taksonomi-filogenetik.html#cb114-4" aria-hidden="true" tabindex="-1"></a>finalplot <span class="ot">&lt;-</span> <span class="fu">msaplot</span>(p, dnanih, <span class="at">offset =</span> <span class="fl">0.08</span>, <span class="at">width =</span> <span class="dv">2</span>)<span class="sc">+</span></span>
<span id="cb114-5"><a href="taksonomi-filogenetik.html#cb114-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_viridis_d</span>(<span class="at">alpha =</span> <span class="fl">0.8</span>)</span></code></pre></div>
<pre><code>## Scale for &#39;fill&#39; is already present. Adding another scale for &#39;fill&#39;, which
## will replace the existing scale.</code></pre>
<div class="sourceCode" id="cb116"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb116-1"><a href="taksonomi-filogenetik.html#cb116-1" aria-hidden="true" tabindex="-1"></a>finalplot</span></code></pre></div>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-22-1.png" width="672" /></p>
<p>Menyimpan plot</p>
<div class="sourceCode" id="cb117"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb117-1"><a href="taksonomi-filogenetik.html#cb117-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(</span>
<span id="cb117-2"><a href="taksonomi-filogenetik.html#cb117-2" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;finalplot.png&quot;</span>,</span>
<span id="cb117-3"><a href="taksonomi-filogenetik.html#cb117-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">plot =</span> <span class="fu">last_plot</span>(),</span>
<span id="cb117-4"><a href="taksonomi-filogenetik.html#cb117-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">device =</span> <span class="cn">NULL</span>,</span>
<span id="cb117-5"><a href="taksonomi-filogenetik.html#cb117-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">path =</span> <span class="cn">NULL</span>,</span>
<span id="cb117-6"><a href="taksonomi-filogenetik.html#cb117-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">scale =</span> <span class="dv">1</span>,</span>
<span id="cb117-7"><a href="taksonomi-filogenetik.html#cb117-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">width =</span> <span class="cn">NA</span>,</span>
<span id="cb117-8"><a href="taksonomi-filogenetik.html#cb117-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">height =</span> <span class="cn">NA</span>,</span>
<span id="cb117-9"><a href="taksonomi-filogenetik.html#cb117-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">units =</span> <span class="st">&quot;in&quot;</span>,</span>
<span id="cb117-10"><a href="taksonomi-filogenetik.html#cb117-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">limitsize =</span> <span class="cn">TRUE</span>,</span>
<span id="cb117-11"><a href="taksonomi-filogenetik.html#cb117-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">bg =</span> <span class="cn">NULL</span>,</span>
<span id="cb117-12"><a href="taksonomi-filogenetik.html#cb117-12" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>

</div>
</div>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="daftar-pustaka.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/danilyanedo7/praktikumsismikbits/edit/master/02-tears.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/danilyanedo7/praktikumsismikbits/blob/master/02-tears.Rmd",
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
